src/Jackalope/NodeType/NodeProcessor.php

```diff
@@ -262,12 +279,11 @@ private function assertValidProperty($property)
                     $values = array($values);
                 }
                 foreach ($values as $value) {
-                    if (!preg_match(self::VALIDATE_PATH, $value)) {
-                        throw new ValueFormatException(sprintf(
-                            'Value "%s" for PATH property at "%s" is invalid. Segments are separated by / and allowed chars are -a-zA-Z0-9:_',
-                            $value,
-                            $property->getPath()
-                        ));
+                    try {
+                        // making the path absolute also validates the result to be a valid path
+                        PathHelper::absolutizePath($value, $property->getPath());
```

Comment:
yeah, the one in the client was pointless. it had unescaped `.` in it so
did not validate anything. what i do now is checking that this is
potentially a valid path, although implementation specific restrictions
are not handled this way but that is an issue throughout the code, as
PathHelper has static methods and thus can not be extended to add
restrictions.
so imo its better than before and we can solve the issue of proper path
validation at a different day.